tests/test_exit_codes.py

```
@@ -0,0 +1,205 @@
+        assert exc_info.value.code == EXIT_GENERAL_ERROR
+
+    def test_prints_error_to_stderr(self, capsys):
+        with pytest.raises(SystemExit):
+            error_exit(ValueError("bad value"))
+        captured = capsys.readouterr()
+        assert "error: bad value" in captured.err
+        assert captured.out == ""
+
+    def test_debug_mode_prints_traceback(self, capsys):
+        try:
+            raise ValueError("debug test")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)